From 29522e8913d99a0053ec3eca5dca9901256b0afb Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Fri, 20 Jan 2017 14:31:43 +0100
Subject: [PATCH 56/56] fixup! Do not add a trailing slash when converting / to
 a Win32 path

After long discussions, and several attempts to address this on Git's
side (t0060-path-utils failed specifically when this patch was
reverted), finally Git is ready to have this patch dropped.

This is a step in the right direction: it reduces the difference between
the official MSYS2 runtime and Git for Windows' forked version.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/path.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index 56b7ade..9326929 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -733,11 +733,11 @@ path_conv::check (const char *src, unsigned opt,
 	      need_directory = 1;
 	      *--tail = '\0';
 	    }
-	  /* Special case for "/" must not set need_directory, neither remove
+	  /* Special case for "/" must set need_directory, without removing
 	     trailing slash */
 	  else if (tail == path_copy + 1 && isslash (tail[-1]))
 	    {
-	      need_directory = 0;
+	      need_directory = 1;
 	    }
 	  path_end = tail;
 
-- 
2.9.0

